core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -32,6 +32,13 @@ import org.apache.spark.{Logging, SparkContext, SparkEnv, SparkException}
 import org.apache.spark.scheduler.TaskSchedulerImpl
 import org.apache.spark.scheduler.cluster.CoarseGrainedSchedulerBackend
 import org.apache.spark.util.Utils
+import com.google.common.collect.{BiMap, HashBiMap}
+import org.apache.mesos.protobuf.ByteString
+import java.util.concurrent.locks.ReentrantLock
+
+case class SlaveStatus(var executorRunning: Boolean, var taskRunning: Boolean) {
```

Comment:
this should not be public. Also, it would be best if you could put this in a different file where all the messages live. Also, why are these vars? Case classes by definition are supposed to have immutable fields.